packages/emails/email-manager.ts

```diff
@@ -302,6 +302,7 @@ export const sendCancelledEmails = async (
                 attendeeName: attendee.name,
                 host: calendarEvent.organizer.name,
                 eventType: calendarEvent.type,
+                eventDuration: calendarEvent.length as number,
```

Comment:
Why are we doing type assertion here? Isn't length supposed to be a number only?

Type assertions might cause runtime errors, so, they should be avoided unless absolutely necessary.  It is best to have a comment in the code itself as to why the assertion is necessary

Reply:
Because `length` is set as an optional property in the `CalendarEvent` type, and `eventDuration` is a required field there, I added that. If we want to remove the type assertion here, we would need to make `length` a required property and updation needed in many places in the code since `CalendarEvent` is used widely. Let me know what you think.

Reply:
Understood the reason now. But that is a problem that we need to solve, otherwise we could have cases where `eventDuration` would be undefined though you are force telling typescript that it would be a number.

In such a case, we would end up having wrong event name generation.

I think we should atleast try and see how much changes are required, we need to anyway know when that custom event name generation will be wrong.  

If a lot of changes are there, we could also think about logging the scenario when this happens and instead of doing type assertion here, we could intentionally send null here and accept null for eventDuration. There we can check null and handle it and log properly.

Reply:
Hey @hariombalhara, I checked it and I think that if not here, then in other cases, we might need to do type assertion. I found that in some cases, the length is extracted from booking.eventType, and in those cases, the length can be number | null. So, what should we do now? If type assertion is problematic, can we make the duration an optional property instead? I understand that every event has a duration, but still

Reply:
@hariombalhara any thoughts?

Reply:
@hariombalhara any thoughts?

Reply:
Okay, thanks for the commit